### ISSUE TITLE
Unused section warning with warn_unused_configs and cache

### ISSUE THREAD

**Issue by The-Compiler** [python/mypy#5957]
With mypy 0.641 or the latest git master, and a config file like:

```ini
[mypy]
warn_unused_configs = True
[mypy-bdb]
ignore_missing_imports = True
```

checking a file like

```python3
import bdb
```

suppresses the `error: No library stub file for standard library module 'bdb'` error as expected.

However, when running mypy a second time, I assume results are retrieved from its cache, and it shows:

```
Warning: unused section(s) in mypy.ini: [mypy-bdb]
```

which means I can only use `--warn-unused-configs` selectively (like after editing the config), as otherwise, I'd get a lot of bogus warnings in day-to-day usage.

This is probably somewhat similar to #2960, which is the same kind of issue for `--warn-unused-ignores`.

**Comment by petergaultney** [comment]
I see this on 0.700 as well. passing `--no-warn-unused-configs` does silence the error, but it seems strange, since my 
```
[mypy-boto3]
ignore_missing_imports = True
```

is definitely doing something.

**Comment by jtherrmann** [comment]
I'm getting the same problem with 0.701 with these lines in `mypy.ini`:

```
[mypy-tests.*]
disallow_untyped_defs = False
```

They're definitely working, but I still get the `Warning: unused section(s) in mypy.ini: [mypy-tests.*]` error. This happens regardless of whether `.mypy_cache` is present.

Though, oddly enough, I'm doing this in two different projects but only getting this problem in one of them. I haven't had time to do further debugging.

**Comment by JukkaL** [comment]
Increasing priority since this seems to affect quite a few people. I think that I've seen this myself as well.

**Comment by gvanrossum** [comment]
I can do this. Or we can mark it "easy first issue"?

**Comment by JukkaL** [comment]
I suspect that this might be a bit difficult for a first issue, at least without a few hints.

**Comment by gvanrossum** [comment]
I can do a simple version of this: ignore the flag when in incremental mode (which is the default). I'm not sure I'm up to fixing it the way we fixed warn_unused_casts/ignores.
